src/cholesky.jl

```diff
@@ -28,7 +37,9 @@ end
             push!(q.args, :($L_m_n = muladd(-$L_m_k, $L_n_k', $L_m_n)))
         end
         L_n_n = Symbol(:L_,n,:_,n)
-        push!(q.args, :($L_n_n = sqrt($L_n_n)))
+        L_n_n_ltz = Symbol(:L_,n,:_,n,:_,:ltz)
+        push!(q.args, :(_nonpdcheck($L_n_n) || return _chol_failure(A, $n, check)))
```

Comment:
I think `info` is supposed to be `-1` in case of a failure? https://github.com/JuliaLang/julia/blob/36e1839c71bed16db1b0cebe04d37d87097aa366/stdlib/LinearAlgebra/src/cholesky.jl#L289

Reply:
```julia
julia> using StaticArrays, LinearAlgebra
[ Info: Precompiling StaticArrays [90137ffa-7385-5640-81b9-e52037218182]
x =
julia> x = Float64.(1:12);

julia> sx = SVector{12}(x);

julia> x = @SVector rand(4);

julia> s = x * x';

julia> cholesky(Matrix(s),check=false)
Failed factorization of type Cholesky{Float64, Matrix{Float64}}

julia> ans.info
2

julia> cholesky(s,check=false)
Failed factorization of type Cholesky{Float64, SMatrix{4, 4, Float64, 16}}

julia> ans.info
2
```

Reply:
```julia
julia> x = @SMatrix rand(4,2);

julia> s = x * x';

julia> cholesky(Hermitian(s),check=false)
Failed factorization of type Cholesky{Float64, SMatrix{4, 4, Float64, 16}}

julia> ans.info
3

julia> cholesky(Hermitian(Matrix(s)),check=false)
Failed factorization of type Cholesky{Float64, Matrix{Float64}}

julia> ans.info
3
```
It's the diagonal that failed / 1+rank.